### PR TITLE
Use Alire build system

### PR DESCRIPTION
**Pull request by CohenArthur**
- git: Ignore bin directory
- alr: Add basic Alire build system

### CODE REVIEW

**Comment by CohenArthur** [comment]
This PR also adds gnatcoll as a dependency for our project

**Review by n1tram1 — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -1 +1,26 @@
...
+config/
+obj/
+lib/
+alire/
+config/
+obj/
+lib/
+alire/
+config/
+bin/
```
Comment (Skallwar) [inline .gitignore]: Some duplicates I think

**Review by Skallwar — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (CohenArthur) [reply .gitignore]: Ah, good catch. This was auto generated by alire but I had to run it multiple times

Comment (CohenArthur) [reply .gitignore]: Fixed in 6530dc1

**Review by n1tram1 — APPROVED** [review]

**Review by Skallwar — APPROVED** [review]

**Pull request merged**